internal/v2/tls_config_store/tls_config_store.go

```
@@ -22,53 +24,188 @@ var (
+		log.Fatalf("Client: failed to send SessionReq: %v", err)
+	}
+	log.Printf("Client: sent SessionReq")
+
+	// Get the response containing config from S2Av2.
+	resp, err := cstream.Recv()
+	if err != nil {
+		log.Fatalf("Client: failed to recieve SessionResp: %v", err)
+	}
+	log.Printf("Client: recieved SessionResp")
+
+	// TODO(rmehta19): Handle resp.GetStatus().
+
+	// Extract TLS configiguration from SessionResp.
+	tlsConfig := resp.GetGetTlsConfigurationResp().GetServerTlsConfiguration()
+	var certList []tls.Certificate
+	// Fake S2Av2 only puts one cert in CertitificateChain, for loop iterates
+	// once.
+	for _, v := range tlsConfig.CertificateChain {
+		// TODO(rmehta19): Call remote signer library for private key.
+		cert, err := tls.X509KeyPair([]byte(v), serverKey)
+		if err != nil {
+			log.Fatalf("Failed to generate X509KeyPair: %v", err)
+		}
+		certList = append(certList, cert)
 	}
 
 	certPool := x509.NewCertPool()
 	certPool.AppendCertsFromPEM(clientCert)
 
+	// Map S2Av2 TLSVersion to consts defined in tls package.
+	var minVersion uint16
+	var maxVersion uint16
+	switch x := tlsConfig.MinTlsVersion; x {
+	case commonpb.TLSVersion_TLS_VERSION_1_0:
+		minVersion = tls.VersionTLS10
+	case commonpb.TLSVersion_TLS_VERSION_1_1:
+		minVersion = tls.VersionTLS11
+	case commonpb.TLSVersion_TLS_VERSION_1_2:
+		minVersion = tls.VersionTLS12
+	case commonpb.TLSVersion_TLS_VERSION_1_3:
+		minVersion = tls.VersionTLS13
+	default:
+		minVersion = tls.VersionTLS13
+	}
+
+	switch x := tlsConfig.MaxTlsVersion; x {
+	case commonpb.TLSVersion_TLS_VERSION_1_0:
+		maxVersion = tls.VersionTLS10
+	case commonpb.TLSVersion_TLS_VERSION_1_1:
+		maxVersion = tls.VersionTLS11
+	case commonpb.TLSVersion_TLS_VERSION_1_2:
+		maxVersion = tls.VersionTLS12
+	case commonpb.TLSVersion_TLS_VERSION_1_3:
+		maxVersion = tls.VersionTLS13
+	default:
+		maxVersion = tls.VersionTLS13
+	}
+
+	if minVersion > maxVersion {
+		log.Printf("S2Av2 provided minVersion > maxVersion")
```

Please return an error in this case.